### PR TITLE
CONTRIBUTING.md: update testing section

### LINKED ISSUE #5874: "Testing" section of CONTRIBUTING.md needs some love
https://github.com/tendermint/tendermint/blob/master/CONTRIBUTING.md#testing is out of date. It references CircleCI instead of Github actions, and could be improved with information on the E2E testing, Jepsen testing, and MBT. I'm also not sure if the RPC Testing section is still up-to-date.

### PR DESCRIPTION
**Pull request by melekes**
[✌️ RENDERED](https://github.com/tendermint/tendermint/blob/ad5a2ec28b1f3b3ded95c7be0ea5446065199d78/CONTRIBUTING.md)

Closes #5874

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -129,16 +129,16 @@ You should now be able to run `make proto-gen` from inside the root Tendermint d
 
 ### Visual Studio Code
 
-If you are a VS Code user, you may want to add the following to your `.vscode/settings.json`:  
+If you are a VS Code user, you may want to add the following to your `.vscode/settings.json`:
```
Comment (alexanderbez) [inline CONTRIBUTING.md:132]: _Workspace_ settings. I don't think we should have this in a global settings.

Comment (melekes) [reply CONTRIBUTING.md:132]: Are you referring to my Vim stripping whitespaces or vscode settings?

Comment (alexanderbez) [reply CONTRIBUTING.md:132]: Sorry, I misread this!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -247,33 +247,33 @@ Each PR should have one commit once it lands on `master`; this can be accomplish
...
 
-This major release process assumes that this release was preceded by release candidates. 
-If there were no release candidates, and you'd like to cut a major release directly from master, see below. 
+This major release process assumes that this release was preceded by release candidates.
+If there were no release candidates, and you'd like to cut a major release directly from master, see below.
 
 1. Start on the latest RC branch (`RCx/vX.X.0`).
 2. Run integration tests.
 3. Branch off of the RC branch (`git checkout -b release-prep`) and prepare the release:
-   - "Squash" changes from the changelog entries for the RCs into a single entry,
```
Comment (marbar3778) [inline CONTRIBUTING.md:256]: I believe @tessr left these space in for formatting reasons.

Comment (tessr) [reply CONTRIBUTING.md:256]: The leading spaces are important for formatting (and sometimes the trailing spaces are too, honestly - markdown can be a pain). But in this case, I think it's fine to trim these trailing spaces.

**Review by marbar3778 — APPROVED** [review]
LGTM. Nice additions!

**Pull request merged**